var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -283,8 +284,20 @@ def filter_detected_exes(cls, prefix, exes_in_prefix):
             # clang++ matches g++ -> clan[g++]
             if any(x in exe for x in ('clang', 'ranlib')):
                 continue
-            # Filter out links in favor of real executables
-            if os.path.islink(exe):
+            # Filter out links in favor of real executables on
+            # all systems but Cray
+            host_platform = str(spack.architecture.platform())
+            if os.path.islink(exe) and host_platform != 'cray':
+                continue
+            # Apple's gcc is actually apple clang, so skip it.
+            # Users can add it manually to compilers.yaml at their own risk.
+            try:
+                output = spack.compiler.get_compiler_version_output(
+                    exe, '--version'
```

Comment:
Would it make sense to use `Gcc.version_argument` or grab the version with `Gcc.default_version`? That logic will be moved here eventually but while there is still a `Gcc` class I think it makes sense to use it.

Reply:
I'd avoid it. What I'm trying to do here is to extract from #17525 all the improvements to packages that already make sense in `develop`. #17525 will then remove all the version detection logic from `spack.compilers`.

Reply:
Should this use any of the logic from `determine_version` then? Or perhaps this check should move there? There seems to be some overlap between the intent of `determine_version` and `filter_detected_exes`. It might make sense to constrain `filter_detected_exes` to not invoke the executable.

Reply:
I think it's a good idea to move the check in `determine_version`.